robottelo/utils/sso.py

```diff
@@ -0,0 +1,206 @@
+import json
+import random
+from functools import cached_property
+from functools import lru_cache
+
+from box import Box
+from broker.hosts import Host
+from fauxfactory import gen_string
+
+from robottelo.config import settings
+from robottelo.constants import KEY_CLOAK_CLI
+from robottelo.datafactory import valid_emails_list
+
+RHSSO_NEW_USER = Box(
+    email="test_user@example.com",
+    emailVerified="true",
+    enabled="true",
+    firstName="first_name",
+    lastName="last_name",
+    username="random_name",
+)
+
+RHSSO_USER_UPDATE = Box(realm="realm_name", userId="user_id")
+
+RHSSO_NEW_GROUP = Box(
+    name="group_name",
+)
+
+RHSSO_RESET_PASSWORD = Box(temporary="false", type="password", value="")
+
+
+class SSOHost(Host):
```

Comment:
According to our helpers refactor history and best practices, , the utils module is for external tools that helps pytest/robottelo to execute the tests or which are not related to host objects or not satellite endpoint objects etc. 

So If the helpers are associated with host objects as we are having here then the best place for them is `robottelo/hosts.py`. Unfortunately `robottelo/hosts.py` is already a big file containing Chost, Sat and Capsule classes and its methodsssssss! 

So we should just initialise the SSOHost in that file and you should move all SSOHost methods to `SSOHost_mixins.py` categorised in classes and then inherit those classes/methods in SSOHost class. Refer https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L123 where ContentHost class importing ContentHost mixins class. Also I agree with @mshriver comment of inheriting ContentHost class than Host class.

@mshriver @JacobCallahan  We should move all methods of CHost, Sat and Capsule classes methods to the mixins and just initialise classes in hosts.py. Its a separate effort and not blocking this PR for it!

Reply:
`class SSOHost` without 'Satellite' class has no value having the composition of Satellite class with SSOHost class make sense and not need to pass the satellite object in every function. As @mshriver said it is good to have self.satellite attribute associated with rhsso host class

Reply:
Fixes made. After speaking with jake, We decided to keep all contents of ssohost() in hosts.py and not break it up into a mixins file.